src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -571,7 +575,7 @@ private async ValueTask WriteAsyncCore(byte[] buffer, int offset, int count, Can
                         int numWholeBlocksInBytes = numWholeBlocks * _inputBlockSize;
 
                         // Use ArrayPool.Shared instead of CryptoPool because the array is passed out.
-                        byte[] tempOutputBuffer = ArrayPool<byte>.Shared.Rent(numWholeBlocks * _outputBlockSize);
+                        byte[]? tempOutputBuffer = ArrayPool<byte>.Shared.Rent(numWholeBlocks * _outputBlockSize);
```

Comment:
ArrayPool.Rent isn't marked as null-return, why is this a maybe-null local?

Reply:
There is rows below where this is nulled out after rented pool returned